Moonshot/player/PlayerArsenal.gd

```diff
@@ -5,30 +5,210 @@ class LaserBlaster:
+		var shots = []
+		var shot
+		for i in range(no_shots):
+			shot = bullet.instance()
+			shot.f_mag = f_mag
+			shot.damage = damage
+			shot.scale = shot_scale
+			var shot_children = shot.get_children()
+			for shot_child in shot_children:
+				shot_child.scale = shot_scale
+				shot_child.scale = shot_scale
+			shots.append(shot) 
+		return shots
+
+class ShotgunBlaster:
+	var sound = AudioStreamPlayer2D.new()
+	var bullet = load("res://Combat/bullet.tscn")
+	var sfx = load("res://player/assets/sounds/laser_pew.ogg") 
+	var fire_speed = 0.9
+	var f_mag = 800
+	var damage = 0.2
+	var no_shots = 5
+	var shot_rotation_modifiers = [-10, -5, 0, 5, 10]
+	
+	func _init():
+		sfx.set_loop(false)
+		sound.stream = sfx
 
+
+	func shoot():
+		sound.play()
+		var shots = []
+		var shot
+		for i in range(no_shots):
+			shot = bullet.instance()
+			shot.f_mag = f_mag
+			shot.damage = damage
+			shots.append(shot) 
+		return shots
+		
+class TwinShotBlaster:
+	var sound = AudioStreamPlayer2D.new()
+	var bullet = load("res://Combat/bullet.tscn")
+	var sfx = load("res://player/assets/sounds/laser_pew.ogg") 
+	var fire_speed 
+	var f_mag
+	var damage 
+	var no_shots = 2
+	var shot_rotation_modifiers = [-10, 10]
+	
+	func _init():
+		sfx.set_loop(false)
+		sound.stream = sfx
+		# set the attributes to be the same as the LaserBlaster
+		var laser_blaster = LaserBlaster.new()
+		fire_speed = laser_blaster.fire_speed
+		f_mag = laser_blaster.f_mag
+		damage = laser_blaster.damage
+
+	func shoot():
+		sound.play()
+		var shots = []
+		var shot
+		for i in range(no_shots):
+			shot = bullet.instance()
+			shot.f_mag = f_mag
+			shot.damage = damage
+			shots.append(shot) 
+		return shots
+		
+class BigBlaster:
+	var sound = AudioStreamPlayer2D.new()
+	var bullet = load("res://Combat/bullet.tscn")
+	var sfx = load("res://player/assets/sounds/laser_pew.ogg") 
+	var fire_speed = 0.75
+	var f_mag = 600
+	var damage = 0.5
+	var no_shots = 1
+	var shot_scale = Vector2(4, 4)
+	var shot_rotation_modifiers = [0]
+	
+	func _init():
+		sfx.set_loop(false)
+		sound.stream = sfx
+
+
+	func shoot():
+		sound.play()
+		var shots = []
+		var shot
+		for i in range(no_shots):
+			shot = bullet.instance()
+			shot.f_mag = f_mag
+			shot.damage = damage
+			var shot_children = shot.get_children()
+			for shot_child in shot_children:
+				shot_child.scale = shot_scale
+				shot_child.scale = shot_scale
+			shots.append(shot) 
+		return shots
+		
+class BrokenBlaster:
+	var sound = AudioStreamPlayer2D.new()
+	var bullet = load("res://Combat/bullet.tscn")
+	var sfx = load("res://player/assets/sounds/laser_pew.ogg") 
+	var fire_speed 
+	var f_mag 
+	var damage 
+	var no_shots = 1
+	var shot_scale = Vector2(1, 1)
+	var shot_rotation_modifiers = [0]
+	# Shot random spread in angles
+	var shot_spread = 10
+	
+	func _init():
+		sfx.set_loop(false)
+		sound.stream = sfx
+		var laser_blaster = LaserBlaster.new()
+		fire_speed = laser_blaster.fire_speed
+		f_mag = laser_blaster.f_mag
```

Comment:
What does `f_mag` mean? If it is a known thing then this is just for my understanding else can we call `f_mag` `shot_speed` or something a little less arcane for all classes?

Reply:
Yeah it is the speed of the bullet, I used the existing teminology, happy to change it as I had no idea what it meant initially

Reply:
Fair, that makes sense. consistency is king